### DISCUSSION TITLE
Send Buffer data as response from server action

### DISCUSSION THREAD

**Discussion by surnr** [vercel/next.js#67832] [App Router]
I call a server action from client side which will return a json value, inside a key has Buffer data type, I'm getting error that 

```js
Warning: Only plain objects can be passed to Client Components from Server Components. Uint8Array objects are not supported.
  [Uint8Array, ..., ..., ...]
  ```
  
  How can I pass Buffer data from server to client? The buffer data is an Image, which I want to render on client

**Comment by icyJoseph** [comment]
- https://react.dev/reference/rsc/use-server#serializable-parameters-and-return-values
- https://react.dev/reference/rsc/use-client#serializable-types

ArrayBuffer (and TypedArray) is meant to be supported, but, according to, https://github.com/facebook/react/issues/27931, it needed some more work, which has been merged upstream on React. I would try on next@canary and see if it works there.

**Comment by aiherrera** [comment]
Hey guys, any update on this?

**Comment by lamualfa** [comment]
Still doesn't work in latest next.js `15.3.5`:

```bash
Error: [ Server ] Only plain objects can be passed to Client Components from Server Components. Uint8Array objects are not supported.
  {buffer: Uint8Array, vector: ...}
           ^^^^^^^^^^
    at createConsoleError (http://localhost:3000/_next/static/chunks/9bfa0_next_dist_client_f63f4c79._.js:882:71)
    at handleConsoleError (http://localhost:3000/_next/static/chunks/9bfa0_next_dist_client_f63f4c79._.js:1058:54)
    at console.error (http://localhost:3000/_next/static/chunks/9bfa0_next_dist_client_f63f4c79._.js:1223:57)
    at  (<anonymous>)
```

**Edit: Solved**

Use `Uint8Array` instead of `Buffer` will solved the problem.
